### ISSUE TITLE
Feature proposal: detect dead ReLUs

### ISSUE THREAD

**Issue by arodiss** [fchollet/keras#7057]
Some ReLU neurons never fire during entire training epoch due to their unlucky weights. Since they are flat on negative input, their gradient is zero. Hence, their weights are not updated. Hence, they are not trained and are essentially useless.
Additionally, too high learning rate can eventually push neuron's weights into the never-active zone, knocking it out from training.
This problem (often referenced as "dead ReLU") is very common but somewhat tricky to debug because internal state of NN is obscure.
I think it would be nice to have callback that will check for dead ReLU after each training epoch and warn user so he can change initialization scheme, reduce learning rate etc.

I have very crude implementation of the feature here:  https://github.com/fchollet/keras/pull/7056
If community and maintainers agree this makes sense I will finalize this PR

**Comment by nicksam112** [comment]
If y'all are encountering this, you can try using some of the advanced activations like LeakyReLU or PReLU. Otherwise yes I'd agree, it'd be a neat feature

**Comment by arodiss** [comment]
Thank you guys
I will finalize my PR

Leaky ReLU / PReLU are good, as well as recent exponential linear unit, but old plain ReLU is still de-facto standard (and also default value), so make sense to add failproof for it

**Comment by Vakuma** [comment]
Any update regarding this feature?

**Comment by the-moliver** [comment]
https://github.com/farizrahman4u/keras-contrib/blob/13ec4b4174daa5ec8d3817fbc7c3f9bc17f288eb/keras_contrib/callbacks/dead_relu_detector.py

**Comment by arodiss** [comment]
Yes, dead ReLU detector is now available at `keras-contrib`, hence closing this

**Comment by rola93** [comment]
@arodiss It looks great! however, I didn't found any doc about it... I'm newby with Keras, can you give an example? Thankyou!
